### PR TITLE
Reset school password

### PR DESCRIPTION
**Pull request by Mudassar045**

### CODE REVIEW

File: elixir/lib/sarkar/action_handlers/dashboard.ex [hunk 1]
```diff
@@ -549,6 +549,34 @@ defmodule Sarkar.ActionHandler.Dashboard do
...
+			"type" => "RESET_SCHOOL_PASSWORD",
+			"client_id" => _client_id,
+			"payload" => %{
+				"school_id" => school_id,
+				"password" => new_password
+			}
+		},
+	%{id: id, client_id: client_id } = state
+	)do
+		case Sarkar.DB.Postgres.query(
```
Comment (taimur38) [inline elixir/lib/sarkar/action_handlers/dashboard.ex]: this db query should be happening in the auth.ex module, and you should be able to call `Sarkar.Auth.reset_password(school_id, new_password)`

File: dashboard/src/actions/index.ts [hunk 2]
```diff
@@ -208,13 +208,13 @@ export const updateReferralInformation = (school_id: string, value: any) => (dis
...
-	.then((res) => {
-		window.alert(`Update Successful\n${res}`)
-		getReferralsInfo()
-	})
-	.catch(() => {
-		window.alert(`Update Information Failed for school ${school_id}`)
-	})
+		.then((res) => {
+			window.alert(`Update Successful\n${res}`)
+			getReferralsInfo()
```
[the remark this answers is not in the document]
Comment (Mudassar045) [reply dashboard/src/actions/index.ts]: Yes it should be. I didn't wrote this but fix it.

File: web/src/constants/links.ts [hunk 3]
```diff
@@ -7,13 +7,13 @@ type Tutorial = {
...
-		link: "https://www.youtube.com/embed/QN6JzExvbw8?controls=0"
-	},
-	"SCHOOL-LOGIN": {
 		title: "Brief Introduction to MISchool",
 		link: "https://www.youtube.com/embed/SHnVsuqp6G8?controls=0"
 	},
+	"SCHOOL-LOGIN": {
+		title: "",
+		link: ""
+	},
```
[the remark this answers is not in the document]
Comment (Mudassar045) [reply web/src/constants/links.ts]: I forgot to merge staging in this branch

File: web/src/modules/Student/Single/Create/index.tsx [hunk 4]
```diff
@@ -191,7 +191,7 @@ class SingleStudent extends Component<propTypes, S> {
 			const section_id = profile.section_id
 			const class_id = sections.find(section => section.id === section_id).class_id
 
-			if (settings.classes && settings.classes.defaultFee[class_id]) {
+			if (settings.classes && settings.classes.defaultFee && settings.classes.defaultFee[class_id]) {
```
[the remark this answers is not in the document]
Comment (Mudassar045) [reply web/src/modules/Student/Single/Create/index.tsx]: may be from master, I didn't merge staging, might be due to that.

File: dashboard/src/pages/admin/password/index.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+					<input type="text" {...this.former.super_handle(["password"])} placeholder="Enter Password" />
+				</div>
+				<div className="button blue" onClick={this.onRestPassword}>Reset Password</div>
+			</div>
+		</div>
+	}
+}
+
+export default connect((state: RootReducerState) => ({
+	schoolList: state.school_Info.school_list,
```
[the remark this answers is not in the document]
Comment (Mudassar045) [reply dashboard/src/pages/admin/password/index.tsx]: yes I need to call `getSchoolList()` in component did mount

File: elixir/lib/sarkar/auth/auth.ex [hunk 6]
```diff
@@ -166,6 +166,17 @@ defmodule Sarkar.Auth do
...
+		case Sarkar.DB.Postgres.query(
+			Sarkar.School.DB,
+			"UPDATE auth SET password=$2 WHERE id=$1",
+			[school_id, hash(new_password, 52)]) do
+			{:ok, _res} -> {:ok, "Password has been reset successfully"}
+			{:error, err} ->
+				{:ok, "Unable to reset password, Please try again"}
+		end
+	end
+
```
Comment (taimur38) [inline elixir/lib/sarkar/auth/auth.ex]: So now that you are writing the function in this file, you should see that `updatePassword` already exists as a function

Comment (Mudassar045) [reply elixir/lib/sarkar/auth/auth.ex]: yeah that exists. should I use that function or not?

**Pull request merged**